### ISSUE TITLE
webapp meteor/browser-policy can't find a way to set nonce

### ISSUE THREAD

**Issue by ivan133** [meteor/meteor#11031]
### This bug report should include:
- [x] A short, but descriptive title. The title doesn't need "Meteor" in it.
- [x] The version of Meteor showing the problem.
I'm using meteor 1.10.1
- [x] The last version of Meteor where the problem did _not_ occur, if applicable.
Not sure
- [x] The operating system you're running Meteor on.
Ubuntu Linux 18.04
- [x] The expected behavior.
I'm setting nonce in CSP using browser-policy like this
```
import { BrowserPolicy } from 'meteor/browser-policy';
import { Random } from 'meteor/random';
export const nonce = Random.id();
BrowserPolicy.content.allowScriptOrigin(`nonce-${nonce}`);
```
I expect it to be added as is:
"script-src 'self' nonce-zhmhehgjftfs6cmbx 

- [x] The actual behavior.
But meteor/browser-policy  checks if it's starts with http / https and if not it adds it. 
So I get
"script-src 'self' http://nonce-zhmhehgjftfs6cmbx https://nonce-zhmhehgjftfs6cmbx;

Is there a way to set it up without http?

- [ ] A **simple** reproduction!
      (example: A GitHub repository that anyone can clone to observe the problem.)

**Comment by ivan133** [comment]
If it's a wrong place to submit this issue please guide me where is the best place to submit it.
As far as I understand meteor/browser-policy doesn't have standalone repository and a part of webapp package (https://github.com/meteor/meteor/tree/master/packages/webapp)

**Comment by ivan133** [comment]
I guess this is the function to blame 
```
var addSourceForDirective = function (directive, src) {
  if (_.contains(_.values(keywords), src)) {
    cspSrcs[directive].push(src);
  } else {
    var toAdd = [];

    //Only add single quotes to CSP2 script digests
    if (/^(sha(256|384|512)-)/i.test(src)) {
      toAdd.push("'" + src + "'");
    } else {
      src = src.toLowerCase();

      // Trim trailing slashes.
      src = src.replace(/\/+$/, '');

      // If there is no protocol, add both http:// and https://.
      if (! /^([a-z0-9.+-]+:)/.test(src)) {
        toAdd.push("http://" + src);
        toAdd.push("https://" + src);
      } else {
        toAdd.push(src);
      }
    }

    _.each(toAdd, function (s) {
      cspSrcs[directive].push(s);
    });
  }
};
```

**Comment by filipenevola** [comment]
Hi @ivan133 are you willing to write a PR to solve this issue?

**Comment by ivan133** [comment]
For now, I'm using this hack
```
const cachedPolicy = BrowserPolicy.content._constructCsp();
const hackedPolicy = cachedPolicy.replace('script-src \'self\' ', `script-src 'self' 'nonce-${nonce}' `);
BrowserPolicy.content.setPolicy(hackedPolicy);
```

**Comment by ruhci28** [comment]
Hii , can i work on this ?

**Comment by filipenevola** [comment]
hi @ruhci28 sure! Go ahead!

**Comment by namenotrequired** [comment]
Hi @ivan133 , I made a PR to fix it, does this solve it for you? https://github.com/meteor/meteor/pull/11171

**Comment by ivan133** [comment]
@namenotrequired yes! it does!

**Comment by filipenevola** [comment]
The fix will be available on Meteor 1.12.
